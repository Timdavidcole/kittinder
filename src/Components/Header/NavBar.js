import React from "react";
import "./header.css";
import { withRouter } from "react-router-dom";

class NavBar extends React.Component {
  render() {
    const ButtonHome = withRouter(({ history }) => (
      <button
        id="buttonHome"
        type="button"
        onClick={() => {
          history.push("/");
        }}
      >
        Home
      </button>
    ));

    const ButtonFavourites = withRouter(({ history }) => (
      <button
        id="buttonFavourites"
        type="button"
        onClick={() => {
          history.push("/favourites");
        }}
      >
        Favourites
      </button>
    ));
    return (
      <div className={"navBar"}>
        <ButtonHome />
        <ButtonFavourites />
      </div>
    );
  }
}

export default withRouter(NavBar);
