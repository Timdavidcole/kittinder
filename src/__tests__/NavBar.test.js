import NavBar from "../Components/Header/NavBar";
import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

test("checks render and NavBar snapshot", () => {
  const navBar = renderer.create(
    <Router>
      <NavBar />
    </Router>
  );

  let tree = navBar.toJSON();
  expect(tree).toMatchSnapshot();
});

test("clicking Home Button takes to home", async () => {
  const wrapper = await mount(
    <Router>
      <NavBar />
    </Router>
  );
  wrapper.find("#buttonHome").simulate("click");
  expect(location.pathname).toBe("/");
});

test("clicking Favourites Button takes to favourites", async () => {
  const wrapper = await mount(
    <Router>
      <NavBar />
    </Router>
  );
  wrapper.find("#buttonFavourites").simulate("click");
  expect(location.pathname).toBe("/favourites");
});
