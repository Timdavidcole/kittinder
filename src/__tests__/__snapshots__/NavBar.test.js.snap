// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks render and NavBar snapshot 1`] = `
<div
  className="navBar"
>
  <button
    id="buttonHome"
    onClick={[Function]}
    type="button"
  >
    Home
  </button>
  <button
    id="buttonFavourites"
    onClick={[Function]}
    type="button"
  >
    Favourites
  </button>
</div>
`;
